scipy/stats/_continuous_distns.py

```diff
@@ -5872,7 +5872,7 @@ def _stats(self, df, nc, moments='mv'):
         #
         mu, mu2, g1, g2 = None, None, None, None
 
-        gfac = sc.gamma(df/2.-0.5) / sc.gamma(df/2.)
+        gfac = np.exp(sc.gammaln(df/2.-0.5) - sc.gammaln(df/2.))
```

Comment:
Does this change the precision of the calculation for smaller values of `df`?
`df` is an integer and `gamma(df/2)`, `gamma(df/2-0.5)` can, in principle, be computed exactly, so the original ratio computation maintains much of the precision of numerator and denominator.  Even if `sc.gammaln` is quite accurate, taking the `exp` in the proposed computation will tend to reduce the precision.

Reply:
@pvanmulbregt this is a fair point to call out and in this case the actual difference is small, consider: 

```
>>> import scipy.special as sc 
>>> import numpy as np 
>>> df = np.arange(2, 344)
>>> new = np.exp(sc.gammaln(df/2.-0.5) - sc.gammaln(df/2.))
>>> old = sc.gamma(df/2.-0.5) / sc.gamma(df/2.)
>>> 
>>> import matplotlib.pyplot as plt
>>> plt.style.use('seaborn-whitegrid')
>>> diff = old - new
>>> rel_diff = (old - new) / old
>>> 
>>> max(np.abs(diff))
1.787459069646502e-14
>>> max(np.abs(rel_diff))
2.2741286535617743e-13
>>> 
```

and most of the differences are smaller than this at the lower `df` values. 

It doesn't look like github gives the option to include a figure but if you run the lines in listing above and then run

```
plt.plot(df, diff, '-')
plt.show()
```

and similar for `rel_diff` you'll see that it's really not until 150 or so for `df` that we start to see larger differences (for either `diff` or `rel_diff`). 

These seem like small enough numerical differences (e-13) that the numerical precision is not much of a concern, do you agree?

Reply:
Looking at this again in the context of the `nct` the `df>2` in the `_stats` method otherwise the variance does not exist. The calculations above are a bit out of context in that regard so better to disregard the `df=2` value and assume `df>2`. To answer the question though @pvanmulbregt they do not degrade the numerical accuracy for ***valid*** smaller values of `df`.

Reply:
mean exists for df > 1, so df=2 is valid for the mean

Reply:
@josef-pkt yes but in `nct.stats` the 'mv' (mean ***and*** variance) are the defaults

Reply:
But in my original application, I only needed the mean.

I'm currently working with a case with chi2/ncx2 with df=1.6, so maybe I will run into an nct with df=1.6 eventually. :)

But in any case, precision looks good to me and we get a much larger valid range for large df

Reply:
> But in my original application, I only needed the mean.
> 
> I'm currently working with a case with chi2/ncx2 with df=1.6, so maybe I will run into an nct with df=1.6 eventually. :)

Yeah the only potential issue I see is:

https://github.com/scipy/scipy/blob/master/scipy/stats/_continuous_distns.py#L5877

which assumes you're `df>2`, for the `mean` alone (`_stats` is in the callpath of `mean`) this isn't an issue but if `df=2` you'll get a runtime error, even in the mean. This is why I'm  saying 'assumes `df>2`'. 
 
> But in any case, precision looks good to me and we get a much larger valid range for large df

Ok. I think if you've got statsmodels applications w/`df=1.6` then they could have `df=2` or something thereabouts, right? 

I'm thinking we can just wrap `c20` and `c22` calculations in an `if df > 2` block, seems like a simple enough fix to remove the potential runtime warning.

Reply:
`if df > 2` doesn't work because it needs to be vectorized
similar to the where cases